src/Sylius/Bundle/UserBundle/Security/UserLogin.php

```diff
@@ -39,7 +39,7 @@ public function login(UserInterface $user, ?string $firewallName = null): void
         $this->userChecker->checkPostAuth($user);
 
         $token = $this->createToken($user, $firewallName);
-        if (!$token->isAuthenticated()) {
+        if (null === $token->getUser() || [] === $token->getUser()->getRoles()) {
```

Comment:
Why keeping `[] === $token->getUser()->getRoles()` here?

Authorization could still happen and be legit with any voter but the default Symfony `RoleVoter`, given there is a proper user.
(https://github.com/symfony/symfony/blob/5.4/src/Symfony/Component/Security/Core/Authentication/Token/TokenInterface.php#L74).

Reply:
I've added that because of this current spec => https://github.com/Sylius/Sylius/blob/fa4c1d30dd9f94f4c8bb4e80b8fb3208fdf38c62/src/Sylius/Bundle/UserBundle/spec/Security/UserLoginSpec.php#L73

But maybe we could remove it.

Reply:
as we have this test, lets keep it